collect_app/src/main/java/org/odk/collect/android/receivers/NetworkReceiver.java

```diff
@@ -58,28 +60,51 @@ public void onReceive(Context context, Intent intent) {
         String action = intent.getAction();
         ConnectivityManager manager = (ConnectivityManager) context.getSystemService(
                 Context.CONNECTIVITY_SERVICE);
+
         NetworkInfo currentNetworkInfo = manager.getActiveNetworkInfo();
 
-        if (action.equals(ConnectivityManager.CONNECTIVITY_ACTION)) {
+        if (action.equals(ConnectivityManager.CONNECTIVITY_ACTION)
+                || action.equals("org.odk.collect.android.FormSaved")) {
             if (currentNetworkInfo != null
                     && currentNetworkInfo.getState() == NetworkInfo.State.CONNECTED) {
-                uploadForms(context, isFormAutoSendOptionEnabled(currentNetworkInfo));
-            }
-
-            ServerPollingJob.pollServerIfNeeded();
-        } else if (action.equals("org.odk.collect.android.FormSaved")) {
-            ConnectivityManager connectivityManager = (ConnectivityManager) context
-                    .getSystemService(Context.CONNECTIVITY_SERVICE);
-            NetworkInfo ni = connectivityManager.getActiveNetworkInfo();
+                autoSendInstances(context, networkTypeMatchesAutoSendSetting(currentNetworkInfo));
 
-            if (ni != null && ni.isConnected()) {
-                uploadForms(context, isFormAutoSendOptionEnabled(currentNetworkInfo));
+                // If we just got connectivity, poll the server for form updates if we are due for it
+                if (action.equals(ConnectivityManager.CONNECTIVITY_ACTION)) {
+                    ServerPollingJob.pollServerIfNeeded();
```

Comment:
We can move it out of this nested if statement since we check connection in [ServerPollingJob](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/tasks/ServerPollingJob.java#L193).

Reply:
I mean out of https://github.com/opendatakit/collect/blob/b01d097e3efbbc13e852663cbb839d5e1df4ce84/collect_app/src/main/java/org/odk/collect/android/receivers/NetworkReceiver.java#L66